functests/1_performance/cpu_management.go

```diff
@@ -324,6 +326,52 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 				}
 			}
 		})
+
+		It("should place the pod entry point in the container housekeeping CPUs", func() {
+			// TODO: ensure the CI machine have 4 exclusive cores available
+			// WARNING: make sure the entry point of the test pod is
+			//          a single-threaded program - /bin/sleep is fine.
+			testpod = getTestPodWithAnnotations(4, annotations)
+
+			By("Starting the pod")
+			err := testclient.Client.Create(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Getting the container cpuset.cpus cgroup")
+			containerID, err := pods.GetContainerIDByName(testpod, "test")
+			Expect(err).ToNot(HaveOccurred())
+
+			cmd := []string{"/bin/bash", "-c", fmt.Sprintf("find /rootfs/sys/fs/cgroup/cpuset/ -name *%s*", containerID)}
+			containerCgroup, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking what CPU the pod is allowed to run into")
+			cmd = []string{"/bin/bash", "-c", fmt.Sprintf("cat %s/cpuset.cpus", containerCgroup)}
+			output, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			cpus, err := cpuset.Parse(output)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking on which processor the container entry point is running")
+			// see: https://unix.stackexchange.com/questions/79520/determining-the-particular-processor-on-which-a-process-is-running
+			podCmd := []string{"/bin/bash", "-c", "/bin/cut -d' ' -f39 < /proc/1/stat"}
+			podOutput, err := pods.ExecCommandOnPod(testpod, podCmd)
+
+			psr, err := strconv.Atoi(strings.TrimSpace(string(podOutput)))
+			Expect(err).ToNot(HaveOccurred())
+
+			cpuIDs := cpus.ToSlice()
+			houseKeepingCPU := cpuIDs[0] // TODO this is not correct, also cpuIDs[0] thread sibling is actually OK
+			Expect(psr).To(Equal(houseKeepingCPU), fmt.Sprintf("pod entry point is running on cpu %d - housekeeping cpu = %d", psr, houseKeepingCPU))
+
+			By("Deleting the pod")
+			deleteTestPod(testpod)
```

Comment:
If the test will fail, will the pod will be deleted anyway?

Reply:
Yes, because we have the `AfterEach` above. Now I'm trying to think about the less confusing behaviour, though.